Automattic/wp-calypso
# Recommended plugins: add data layer

#### Changes proposed in this Pull Request

(Commandeered by @jsnmoon)

Works with the new `/sites/<slug>/plugins/recommended` API endpoint introduced in D24342-code by @skjaini.

<img width="1062" alt="Screen Shot 2019-03-13 at 12 59 48 PM" src="https://user-images.githubusercontent.com/4044428/54306836-e835f180-458f-11e9-8707-6a2b2922fcb3.png">

This adds a feature flag that replaces the `Featured` section with a dynamically curated `Recommended` section for the plugins page (`/plugins/<slug>`). Please note that the current implementation **only supports Jetpack connected sites with at least one plugin enabled**. This feature flag is enabled by default for wpcalypso and development environments.

This PR introduces:
- [x] a tested data layer implementation for the recommended plugins network request
- [x] a tested Redux state implementation (actions/reducer/selector)
- [x] a query component that dispatches a network request action upon mounting
- [x] data layer to do the API fetch

#### Testing instructions

1. `npm install && npm run test-client`; ensure all tests pass (including new ones).
2. `npm start`, navigate to the [plugins page](http://calypso.localhost:3000/plugins/), and select a Jetpack site using the Site Switcher on the left sidebar. (Alternatively, you can opt for [Calypso live branch](https://calypso.live/plugins?branch=add/recommended-plugins-data-layer) instead of a local installation.)
3. Ensure that the recommended plugins load as expected on the top section of the plugins page. (Due to how we check if a site is a candidate for plugin recommendations, you may see a flash of Featured plugins before Recommended plugins can load.)
